src/test/java/bio/terra/service/dataset/DatasetRequestValidatorTest.java

```diff
@@ -233,6 +251,35 @@ public void testInvalidColumnMode() throws Exception {
         errorModel, new String[] {"InvalidPrimaryKey", "InvalidColumnMode", "InvalidColumnMode"});
   }
 
+  /**
+   * Modify and return the request so that it has a single table with a single column of the
+   * specified name, no relationships, and no assets.
+   */
+  private DatasetRequestModel withNamedColumn(DatasetRequestModel request, String columnName) {
+    ColumnModel column = new ColumnModel().name(columnName).datatype(TableDataType.STRING);
+    TableModel table = new TableModel().name("table").columns(List.of(column));
+    request.getSchema().tables(List.of(table)).relationships(List.of()).assets(List.of());
+
+    return request;
+  }
+
+  @Test
+  public void testInvalidColumnName() throws Exception {
+    DatasetRequestModel req = buildDatasetRequest();
+
+    // Column names with leading numbers are invalid
+    checkValidationErrorModel(
+        expectBadDatasetCreateRequest(withNamedColumn(req, "1")), new String[] {"Pattern"});
```

Comment:
Would it make sense to use a value that's not _just_ a 1 but having it be leading (like "1column" or "_column" for the next check too) since I could see disallowing all number (or all underscores) as being it's own thing

Reply:
That's a good point, I'll update!